numba/types/containers.py

```diff
@@ -308,6 +310,10 @@ def key(self):
 class NamedTuple(_HeterogeneousTuple, BaseNamedTuple):
 
     def __init__(self, types, cls):
+
+        if not isinstance(types, Iterable):
+            raise TypingError("{} type is not iterable".format(type(types)))
```

Comment:
I think this check would be better placed in the `_HeterogeneousTuple` class as a method and then it be called by the `__init__` of both `Tuple` and `NamedTuple` as they both require an `Iterable` for the `types` argument. Further this error message perhaps should read:
```
Argument 'types' must be an Iterable.
```
so as to explain what the requirement is and that it relates to the `types` argument.

Following this tests for all the `*Tuple*` constructors accepting an `Iterable` types can be written.

Reply:
For `Tuple` I had to put the check on `__new__` since there is a conditional there using `types`.